lib/src/lds/df1/efdg2.dart

```diff
@@ -19,4 +33,162 @@ class EfDG2 extends DataGroup {
 
   @override
   int get tag => TAG.value;
+
+  Uint8List? photoData;
+  int? _photoDataType;
+
+  PhotoType? get photoType {
+    if (_photoDataType == null) return null;
+
+    return _photoDataType == 0 ? PhotoType.jpeg : PhotoType.jpeg2000;
+  }
+
+  @override
+  void parse(Uint8List content) {
+    final tlv = TLV.fromBytes(content);
+    if (tlv.tag != tag) {
+      throw EfParseError(
+          "Invalid DG2 tag=${tlv.tag.hex()}, expected tag=${TAG.value.hex()}");
+    }
+
+    final data = tlv.value;
+    final bigt = TLV.decode(data);
+
+    if (bigt.tag.value != BIOMETRIC_INFORMATION_GROUP_TEMPLATE_TAG) {
+      throw EfParseError(
+          "Invalid object tag=${bigt.tag.value.hex()}, expected tag=$BIOMETRIC_INFORMATION_GROUP_TEMPLATE_TAG");
+    }
+
+    final bict = TLV.decode(bigt.value);
+
+    if (bict.tag.value != 0X02) {
+      throw EfParseError(
+          "Invalid object tag=${bict.tag.value.hex()}, expected tag=${0X02}");
+    }
+
+    int bitCount = (bict.value[0] & 0xFF);
+
+    for (var i = 0; i < bitCount; i++) {
+      _readBit(bigt.value.sublist(bict.encodedLen), i);
+    }
+  }
+
+  _readBit(Uint8List stream, int index) {
+    final tvl = TLV.decode(stream);
+
+    if (tvl.tag.value != BIOMETRIC_INFORMATION_TEMPLATE_TAG) {
+      throw EfParseError(
+          "Invalid object tag=${tvl.tag.value.hex()}, expected tag=${BIOMETRIC_INFORMATION_TEMPLATE_TAG}");
+    }
+
+    var bht = TLV.decode(tvl.value);
+
+    if (bht.tag.value == SMT_TAG) {
+      _readStaticallyProtectedBIT();
+    } else if ((bht.tag.value & 0xA0) == 0xA0) {
+      var sbh = _readBHT(tvl.value);
+
+      var bdb = _decodeSMTValue();
+
+      var faceInfo = _readBiometricDataBlock(sbh);
+    }
+  }
+
+  _readStaticallyProtectedBIT() {}
+
+  List<DecodedTV> _readBHT(Uint8List stream) {
+    final bht = TLV.decode(stream);
+    int bhtLength = stream.length;
+    int bytesRead = bht.encodedLen;
+    var elements = <DecodedTV>[];
+    while (bytesRead < bhtLength) {
+      final tlv = TLV.decode(stream.sublist(bytesRead));
+      bytesRead += tlv.encodedLen;
+      elements.add(tlv);
+    }
+
+    return elements;
+    // const expectedBHTTag =
+    //     (BIOMETRIC_HEADER_TEMPLATE_BASE_TAG /* + index */) & 0xFF;
+    // if (tvl.tag.value != expectedBHTTag) {
+    //   throw EfParseError(
+    //       "Invalid object tag=${tvl.tag.value.hex()}, expected tag=${BIOMETRIC_INFORMATION_TEMPLATE_TAG}");
+    // }
+
+    // var bht = TLV.decode(tvl.value);
+  }
+
+  _decodeSMTValue() {}
+
+  _readBiometricDataBlock(List<DecodedTV> sbh) {
+    final tlv = TLV.decode(sbh.first.value);
+
```

Comment:
Add check to match tag to either `0x5F2E - BIOMETRIC_DATA_BLOCK_TAG` or `0x7F2E - BIOMETRIC_DATA_BLOCK_CONSTRUCTED_TAG`